aviary/examples/external_subsystems/engine_NPSS/table_engine_builder.py

```diff
@@ -0,0 +1,186 @@
+            Inputs to Aviary model.
+
+        Returns
+        -------
+        prob : openmdao.core.Group
+            engine model for off-design.
+        '''
+        interp_method = aviary_inputs.get_val(Aircraft.Engine.INTERPOLATION_METHOD)[0]
+
+        # interpolator object for engine data
+        engine = om.MetaModelSemiStructuredComp(
+            method=interp_method, extrapolate=True, vec_size=num_nodes, training_data_gradients=True)
+
+        csv_path = pkg_resources.resource_filename(
+            "aviary",
+            os.path.join("examples", "external_subsystems", "engine_NPSS",
+                         "NPSS_Model", "Output",  "RefEngine.outputAviary")
+        )
+
+        engine_data = np.genfromtxt(csv_path, skip_header=0)
+
+        # Sort the data by Mach, then altitude, then throttle
+        engine_data = engine_data[np.lexsort(
+            (engine_data[:, 2], engine_data[:, 1], engine_data[:, 0]))]
+
+        zeros_array = np.zeros((engine_data.shape[0], 1))
+        # create a new array for thrust_max. here we take the values where throttle=1.0
+        thrust_max = np.zeros((engine_data.shape[0], 1))
+
+        # for a given mach, altitude, and hybrid throttle setting, the thrust_max is the value where throttle=1.0
+        for i in range(engine_data.shape[0]):
+            # find the index of the first instance where throttle=1.0
+            index = np.where((engine_data[:, 0] == engine_data[i, 0]) & (
+                engine_data[:, 1] == engine_data[i, 1]) & (engine_data[:, 2] == 1.0))[0][0]
+            thrust_max[i] = engine_data[index, 3]
+
+        print(Dynamic.Mission.THRUST, '--------------------------------------')
+
+        # add inputs and outputs to interpolator
+        engine.add_input(Dynamic.Mission.MACH,
+                         engine_data[:, 0],
+                         units='unitless',
+                         desc='Current flight Mach number')
+        engine.add_input(Dynamic.Mission.ALTITUDE,
+                         engine_data[:, 1],
+                         units='ft',
+                         desc='Current flight altitude')
+        engine.add_input(Dynamic.Mission.THROTTLE,
+                         engine_data[:, 2],
+                         units='unitless',
+                         desc='Current engine throttle')
+        engine.add_output(Dynamic.Mission.THRUST,
+                          engine_data[:, 3],
+                          units='lbf',
+                          desc='Current net thrust produced')
+        engine.add_output(Dynamic.Mission.THRUST_MAX,
+                          thrust_max,
+                          units='lbf',
+                          desc='Max net thrust produced')
+        engine.add_output(Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
+                          -engine_data[:, 4],
+                          units='lbm/s',
+                          desc='Current fuel flow rate ')
+        engine.add_output(Dynamic.Mission.ELECTRIC_POWER,
```

Comment:
So, i see electric power, nox_rate, and T4 on here. I assume those are things that NPSS can calculate, but isn't in this current version of the wrapper? We are only connecting the thrust, max thrust, and neg fuel flow to this metamodel. We should comment out or remove the remaining 3 unused variables, because if we don't, then it is actually interpolating those values on an array of zeros, and probably takes twice as long because of it.

Reply:
Yes, these are not being populated, however Aviary expects this values from an engine, so they need to exist.  Without these outputs there are errors.

Reply:
Interesting. I didn't have to add additional inputs this last year, but Aviary could have changed since then. I will have to investigate.